crates/turbopack/src/lib.rs

```diff
@@ -122,9 +122,7 @@ async fn apply_module_type(
                 source,
                 context.into(),
                 Value::new(EcmascriptModuleAssetType::Ecmascript),
-                *transforms,
                 Value::new(*options),
-                context.compile_time_info(),
```

Comment:
ModuleType need to change to something like:

```
pub enum ModuleType {
    Ecmascript {
        options: EcmascriptOptions,
    },
    Typescript {
        options: EcmascriptOptions,
    },
    TypescriptWithTypes {
        options: EcmascriptOptions,
    },
    TypescriptDeclaration {
        options: EcmascriptOptions,
    },
```

You also need to derive TraceRawVcs for `EcmascriptOptions`